core/src/main/java/net/consensys/eventeum/integration/eventstore/db/MongoEventStore.java

```diff
@@ -105,6 +106,26 @@ public Optional<MessageDetails> getLatestMessageFromTopic(String nodeName, Strin
         return result != null ? Optional.of(result) : Optional.empty();
     }
 
+    @Override
+    public Optional<ContractEventDetails> getContractEvent(
+            String eventSignature,
+            String contractAddress,
+            String blockHash,
+            String transactionHash,
+            BigInteger logIndex
+    ) {
+        final Query query = new Query(Criteria
+                .where("eventSignature").is(eventSignature)
+                .and("address").is(contractAddress)
+                .and("blockHash").is(blockHash)
+                .and("transactionHash").is(transactionHash)
+                .and("logIndex").is(logIndex))
+                .with(Sort.by(Direction.DESC, "timestamp"))
+                .collation(Collation.of("en").numericOrderingEnabled());
+        final ContractEventDetails result = mongoTemplate.findOne(query, ContractEventDetails.class);
+        return result != null ? Optional.of(result) : Optional.empty();
```

Comment:
this line can be replaces by "return Optional.ofNullable(result);"

Reply:
That's right, I just changed it